src/IO.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Marcosh\LamPHPda;
+
+use Marcosh\LamPHPda\HK\HK1;
+use Marcosh\LamPHPda\Instances\IO\IOApplicative;
+use Marcosh\LamPHPda\Instances\IO\IOApply;
+use Marcosh\LamPHPda\Instances\IO\IOFunctor;
+use Marcosh\LamPHPda\Instances\IO\IOMonad;
+use Marcosh\LamPHPda\Typeclass\Applicative;
+use Marcosh\LamPHPda\Typeclass\Apply;
+use Marcosh\LamPHPda\Typeclass\DefaultInstance\DefaultMonad;
+use Marcosh\LamPHPda\Typeclass\Functor;
+use Marcosh\LamPHPda\Typeclass\Monad;
+use Marcosh\LamPHPda\Brand\IOBrand;
+
+/**
+ * @template A
+ * @implements DefaultMonad<IOBrand, A>
+ */
+final class IO implements DefaultMonad
+{
+    /**
+     * @var callable
+     */
+    private $unsafe;
+
+    private function __construct(callable $f)
+    {
+        $this->unsafe = $f;
+    }
+
+    /**
+     * @template B
+     * @param B $x
+     * @return IO<B>
+     */
+    public static function of($x): IO
+    {
+        return new self(static fn () => $x);
+    }
+
+    /**
+     * @template B
+     * @param HK1<IOBrand, B> $b
+     * @return IO<B>
+     *
+     * @psalm-pure
+     */
+    public static function fromBrand(HK1 $b): IO
+    {
+        /** @var IO $b */
+        return $b;
+    }
+
+    /**
+     * @template B
+     * @param Monad<IOBrand> $monad
+     * @param callable(A): HK1<IOBrand, B> $f
+     * @return IO<B>
+     */
+    public function ibind(Monad $monad, callable $f): IO
```

Comment:
please keep the order we have in the other monads, i.e. first `map`, then `apply`, `pure` and `bind`

Reply:
Done.